data-products/src/main/scala/org/sunbird/analytics/model/report/ETBMetricsModel.scala

```diff
@@ -0,0 +1,108 @@
+
+// Textbook ID, Medium, Grade, Subject, Textbook Name, Created On, Last Updated On, Total No of QR Codes, Number of QR codes with atleast 1 linked content,	Number of QR codes with no linked content, Term 1 QR Codes with no linked content, Term 2 QR Codes with no linked content
+case class dceTextbookReport(slug: String, identifier: String, name: String, medium: String, gradeLevel:String, subject: String,
+                               createdOn: String, lastUpdatedOn: String, totalQRCodes: Integer, contentLinkedQR: Integer,
+                               withoutContentQR: Integer, withoutContentT1: Integer, withoutContentT2: Integer, reportName: String)
+
+case class FinalOutput(identifier: String, etb: Option[etbTextbookReport], dce: Option[dceTextbookReport]) extends AlgoOutput with Output
+
+
+object ETBMetricsModel extends IBatchModelTemplate[Empty,Empty,FinalOutput,FinalOutput] with Serializable {
+  implicit val className: String = "org.sunbird.analytics.model.report.ETBMetricsModel"
+  override def name: String = "ETBMetricsModel"
+
+  override def preProcess(events: RDD[Empty], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[Empty] = {
+    sc.emptyRDD
+  }
+
+  override def algorithm(events: RDD[Empty], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[FinalOutput] = {
+    generateReports(config)
+  }
+
+  override def postProcess(events: RDD[FinalOutput], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[FinalOutput] = {
+    implicit val sqlContext = new SQLContext(sc)
+    import sqlContext.implicits._
+
+    if(events.count() > 0) {
+      val configMap = config("reportConfig").asInstanceOf[Map[String, AnyRef]]
+      val reportConfig = JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(configMap))
+
+      val etbTextBookReport = events.map(report => {
+        if(report.etb.size!=0) report.etb.get else etbTextbookReport("","","","","","","","","",0,0,0,0,0,"")
+      }).filter(textbook=>textbook.identifier!="")
+
+      val dceTextBookReport = events.map(report => {
+        if(report.dce.size!=0) report.dce.get else dceTextbookReport("","","","","","","","",0,0,0,0,0,"")
+      }).filter(textbook=>textbook.identifier!="")
+
+      reportConfig.output.map { f =>
+        val etbDf = etbTextBookReport.toDF()
+        CourseUtils.postDataToBlob(etbDf,f,config)
+
+        val dceDf = dceTextBookReport.toDF()
+        CourseUtils.postDataToBlob(dceDf,f,config)
+      }
+
+    } else {
+      JobLogger.log("No data found", None, Level.INFO)
+    }
+    events
+  }
+
+  def generateReports(config: Map[String, AnyRef])(implicit sc: SparkContext): (RDD[FinalOutput]) = {
+    implicit val httpClient = RestUtil
+    val textBookInfo = TextBookUtils.getTextBooks(config, httpClient)
+    val tenantInfo = getTenantInfo(httpClient)
+    TextBookUtils.getTextbookHierarchy(textBookInfo, tenantInfo, httpClient)
```

Comment:
When the httpClient is declared as implicit, there is no need to pass it as parameter or the implicit declaration is not needed

Reply:
Done